langchain4j-vertex-ai/src/main/java/dev/langchain4j/model/vertexai/VertexAiEmbeddingModel.java

```diff
@@ -46,32 +59,48 @@
  */
 public class VertexAiEmbeddingModel implements EmbeddingModel {
 
-    private static final int BATCH_SIZE = 250; // Vertex AI has a limit of up to 250 input texts per request
-
+    private static final int COMPUTE_TOKENS_MAX_INPUTS_PER_REQUEST = 2048;
+    private static final int DEFAULT_MAX_SEGMENTS_PER_BATCH = 250;
+    private static final int DEFAULT_MAX_TOKENS_PER_BATCH = 20_000;
     private final PredictionServiceSettings settings;
+    private final LlmUtilityServiceSettings llmUtilitySettings;
     private final EndpointName endpointName;
     private final Integer maxRetries;
+    private final Integer maxSegmentsPerBatch;
+    private final Integer maxTokensPerBatch;
 
     public VertexAiEmbeddingModel(String endpoint,
                                   String project,
                                   String location,
                                   String publisher,
                                   String modelName,
-                                  Integer maxRetries) {
+                                  Integer maxRetries,
+                                  Integer maxSegmentsPerBatch,
+                                  Integer maxTokensPerBatch) {
+
+        this.endpointName = EndpointName.ofProjectLocationPublisherModelName(
+            ensureNotBlank(project, "project"),
+            ensureNotBlank(location, "location"),
+            ensureNotBlank(publisher, "publisher"),
+            ensureNotBlank(modelName, "modelName")
+        );
+
         try {
             this.settings = PredictionServiceSettings.newBuilder()
                     .setEndpoint(ensureNotBlank(endpoint, "endpoint"))
                     .build();
+
+            this.llmUtilitySettings = LlmUtilityServiceSettings.newBuilder()
+                    .setEndpoint(settings.getEndpoint())
+                    .build();
         } catch (IOException e) {
             throw new RuntimeException(e);
         }
-        this.endpointName = EndpointName.ofProjectLocationPublisherModelName(
-                ensureNotBlank(project, "project"),
-                ensureNotBlank(location, "location"),
-                ensureNotBlank(publisher, "publisher"),
-                ensureNotBlank(modelName, "modelName")
-        );
+
         this.maxRetries = getOrDefault(maxRetries, 3);
+
+        this.maxSegmentsPerBatch = getOrDefault(maxSegmentsPerBatch, DEFAULT_MAX_SEGMENTS_PER_BATCH);
```

Comment:
Good question.
I can definitely check they are positive, that's a good idea. I'll add that.
But then, yes, I hesitated to check the boundaries, but since they can change (like when we went from 5 to 250) so I think we should not check ourselves and let the underlying API complain should another change happen.

Reply:
Added positivity check